app/controllers/api/tags_controller.rb

```diff
@@ -42,7 +42,8 @@ def delete_resource(_type, id, _data = {})
     private
 
     def fetch_category(data)
-      category_id = parse_id(data, :categories) || parse_by_attr(data, :categories, %w(name))
+      category_id = parse_id(data, :categories)
+      category_id ||= collection_class(:categories).find_by_name(data["name"]).id if data["name"]
```

Comment:
I am not quite sure I follow you. This PR *does* change the behavior.

```
{
  "action" : "create",
  "resource" : {
    "category" : { "name" : "bad_lifecycle" },
    "name" : "sub_tag_1_a3",
    "description" : "Sub Tag 1 a3"
  }
}
```

Before:

```
{
  "error": {
    "kind": "bad_request",
    "message": "Category id, href or name needs to be specified for creating a new tag resource",
    "klass": "Api::BaseController::BadRequestError"
  }
}
```

With this PR:

```
{
  "error": {
    "kind": "internal_server_error",
    "message": "undefined method `id' for nil:NilClass",
    "klass": "NoMethodError"
  }
}
```

with an ugly stack trace in the api.log